src/apatheia/jobs.nim

```diff
@@ -16,75 +17,172 @@ logScope:
 
 ## This module provides a simple way to submit jobs to taskpools
 ## and getting a result returned via an async future.
-## 
+##
+## Any compatible arguments of `seq[T]` or `string` args passed
+## via the `submit` macro will be converted into the special `OpenArrayHolder[T]` type.
+## The `submit` macro converts these arguments in this object and retains the 
+## memory associated with the original `seq[T]` or `string` object.
+## This greatly simplifies the passing of these these types in `refc`.
+##
+## Note, for `arc` or `orc` GC's this setup will be replaced with a move operation in the future.
+## These GC's also allow greater support for moving GC types across thread boundaries.
+##
+## Currently this module limits support for GC types to ensure `refc` safety.
 ##
 
 type
-  JobId* = uint ## job id, should match `future.id()`
-
-  JobQueue*[T] = ref object
-    ## job queue object
+  JobQueue*[T] = ref object ## job queue object
     queue*: SignalQueue[(JobId, T)]
     futures*: Table[JobId, Future[T]]
     taskpool*: Taskpool
     running*: bool
 
-  JobResult*[T] = object
-    ## hold a job result to be returned by jobs
+  JobResult*[T] = object ## hold the result of a job after it finishes
     id*: JobId
     queue*: SignalQueue[(JobId, T)]
 
+  OpenArrayHolder*[T] = object
+    data*: ptr UncheckedArray[T]
+    size*: int
+
+  SupportedSeqTypes* = byte | SomeInteger | SomeFloat
```

Comment:
Why are we limited to those? I thought any non-gc type would work?

Reply:
> Why are we limited to those? I thought any non-gc type would work?

Primarily because it's harder to make a type check that is any non-gc types. If you allow an object, it could have fields which contain GC objects, etc. 

So for now, just simple types. We could add support for some object types as needed.